docs/content/any/project/changelogs/2021-11-09.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Release 2021-11-09
+menuTitle: "2021-11-09"
+any: true
+description: >-
+  Changelog for Release 2021-11-09 (0.23.0) containing new features,
+  bug fixes, and more.
+---
+
+## `oso` 0.23.0
+
+### Core
+
+#### Breaking changes
+
+{{% callout "Warning" "orange" %}}
+  This release contains breaking changes. Be sure to follow migration steps
+  before upgrading.
+{{% /callout %}}
+
+Data filtering now treats accessing an undeclared field as an error. All fields
+used in data filtering queries must now be registered ahead of time by including
+them in the `fields` parameter of the `register_class` Oso API function, which
+previously was only necessary for data relations. For example, to use data filtering
+with the rule
+
+```polar
+allow(user, _, foo: Foo) if foo.user_name = user.name;
+```
+
+`user_name` must be included in the `register_class` call for `Foo`:
+
+```py
+# an example in Python
+oso.register_class(Foo, fields={'user_name': str })
+```
+
+This change *only* affects data filtering. Other Oso APIs require no new configuration.
+
+#### Other bugs & improvements
+
+- Fixed a bug where a negated constraint on a dot lookup could cause Polar to crash
+  when the underlying variable became bound.
+- Removed syntax for parenthesized specializers like `f(_: (x));`, which don't
+  currently achieve anything.
+
+### Python
+
+#### Other bugs & improvements
+- Thanks to [Clara McCreery](https://github.com/chmccreery) for a correction to our
+  Python data filtering docs!
+
+#### Platform support
+
+We now publish wheels for musl-based Linux distributions (through the `musllinux`
+tag) and for ARM-based MacOS systems (through the `macosx_11_0_arm64` tag).
+We now publish wheels for Python 3.10 for all supported platforms.
+
+On those systems, you should now be able to use `pip install oso` to get the
+latest Oso package.
+
+### Node.js
+
+#### New features
+- It is now possible to customize the way that types are determined. By passing
+  a custom `isaCheck` parameter to `oso.registerClass`, you can tell Oso how to
+  determine whether a particular value is an instance of that class. This is
+  useful for certain frameworks, like Prisma, which don't use prototype-style
+  classes for objects of different types. In the example below, any object with
+  `type === "user"` will be considered a `User` by the Oso runtime:
+
+```js
+class SomeUserClass {}
+oso.registerClass(SomeUserClass, {
+  name: "User",
+  isaCheck: instance => instance.type === "user"
+});
+```
+
+#### Other bugs & improvements
+- The `Class` type for representing abstract resources for data filtering is
+  now a top-level export.
+
+### Rust
+
+#### Other bugs & improvements
+
+- Oso's Rust REPL no longer swallows its output.
+## `sqlalchemy-oso` 0.23.0
```

Comment:
nit: sometimes in the past we've added `django-oso` and `flask-oso` entries that indicate we bumped the version due to an `oso` version bump. E.g.: https://github.com/osohq/oso/blob/3d3865679be41a702b6c35836b2172d4f86dec90/docs/content/any/project/changelogs/2020-11-10.md?plain=1#L110-L116